onnxruntime/core/providers/webgpu/nn/pool.cc

```diff
@@ -0,0 +1,254 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+#include "core/providers/webgpu/nn/pool.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+namespace {
+
+std::vector<uint32_t> NarrowToU32(const TensorShapeVector& shape) {
+  std::vector<uint32_t> result;
+  result.reserve(shape.size());
+  for (auto dim : shape) {
+    result.push_back(gsl::narrow_cast<uint32_t>(dim));
```

Comment:
please do not use `gsl::narrow_cast` because it is just a `static_cast`. we usually do:
- `onnxruntime::narrow` if want to generate exceptions when the cast exceeded the range
- `static_cast` if it's obviously safe.

Reply:
Done, thanks for the clarification.